### PR TITLE
Initially adding Runtime provisioning interface plus necessary beans

### PR DESCRIPTION
**Pull request by kasundsilva**

### CODE REVIEW

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/RuntimeProvisioningService.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    public void setRuntimeProperties(List<RuntimeProperty> runtimeProperties, String type) throws RuntimeProvisioningException;
+
+    public void updateRuntimeProperties(RuntimeProperty runtimeProperty) throws RuntimeProvisioningException;
+
+    public List<RuntimeProperty> getRuntimeProperties() throws RuntimeProvisioningException;
+
+    public void addCustomDomain(Set<String> domains) throws RuntimeProvisioningException;
+
+    public void updateCustomDomain(String domain) throws RuntimeProvisioningException;
```
Comment (hisamith) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/RuntimeProvisioningService.java]: When updating the custom domain, you may have to pass the new domain as well

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/beans/Query.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.wso2.carbon.appfactory.provisioning.runtime.beans;
+
+/**
+ * Created by kasuns on 12/2/15.
```
Comment (hisamith) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/beans/Query.java]: Update the class comments

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/RuntimeProvisioningService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    public List<String> deployApplication(DeploymentConfig config) throws RuntimeProvisioningException;
+
+    public boolean getDeploymentStatus() throws RuntimeProvisioningException;
+
+    public OutputStream streamRuntimeLogs() throws RuntimeProvisioningException;
+
+    public String getRuntimeLogs(Query query) throws RuntimeProvisioningException;
+
+    public void setRuntimeProperties(List<RuntimeProperty> runtimeProperties, String type) throws RuntimeProvisioningException;
```
Comment (hisamith) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/RuntimeProvisioningService.java]: What is the purpose of the method parameter "type"?

File: modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/beans/Query.java (hunk 2 above)
Comment (kasundsilva) [inline modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/main/java/org/wso2/carbon/appfactory/provisioning/runtime/beans/Query.java]: Will update the class comments ASAP

**Pull request merged**